### DISCUSSION TITLE
Run c7n-org in python script

### DISCUSSION THREAD

**Discussion by gsl-ghassengabsi** [cloud-custodian/cloud-custodian#7825] [AWS]
Hello,
I tried to use c7n_org in a python script to run policies but i had execution errors
this is my script:

import c7n_org
@click.command()
@click.option('-c', '--config', default=config_filename)
@click.option('-u', '--use', default=ec2_filename)
@click.option('-s', default=output_directory)
def config(**configuration):
    json.dumps(configuration)
    print(configuration)
    run_config = c7n_org.cli.Config.empty(**configuration)
    print(run_config)
    c7n_org.cli.run(run_config)`

i had this error Error: Missing option '-c' / '--config'
Can you help e please? Thanks.

**Comment by k-nelis** [comment]
Hello, you can try something like this.

```python
import c7n_org.cli

commands_ex = [
    "--config", "cloudcustodian/c7n-org-config/accounts.yml",
    "--use", "cloudcustodian/aws/policies/s3-check-buckets-public",
    "--output-dir", "output_path",
    "--region", "sa-east-1"
    ]

test = c7n_org.cli

if __name__ == "__main__": 
    ab = test.run.main(args=commands_ex, standalone_mode=True, prog_name="run")
```

**Reply by ajkerrigan** [reply]
Yeah @k-nelis's answer is a good one - the CLI arguments are effectively c7n-org's public API, so running that is going to be the smoothest/safest option.

**Reply by tbatra18** [reply]
@ajkerrigan  @k-nelis @gsl-ghassengabsi - I still get an error, specify region.
Any thoughts ?

**Reply by fdmsantos** [reply]
This code works in my use case. Tested on azure without region flag.

**Reply by nitrocode** [reply]
This looks like a nice way of implementing integration testing of cloud custodian policies. Thanks for sharing
